src/earn/hooks.ts

```diff
@@ -94,3 +92,68 @@ export function useDepositEntrypointInfo({
   }, [asyncExchanges.result])
   return { hasDepositToken, hasTokensOnSameNetwork, hasTokensOnOtherNetworks, canCashIn, exchanges }
 }
+
+export function usePrepareDepositAndWithdrawTransactions(
+  mode: Extract<EarnActiveMode, 'deposit' | 'withdraw' | 'swap-deposit'>
+) {
+  const getTransactionFunction = () => {
+    switch (mode) {
+      case 'deposit':
+      case 'swap-deposit':
+        return prepareDepositTransactions
+      case 'withdraw':
+        return prepareWithdrawTransactionsWithSwap
+      default:
+        throw new Error(`Invalid mode: ${mode}`)
+    }
+  }
+
+  const prepareTransactions = useAsyncCallback(
+    async (args) => {
+      return getTransactionFunction()(args)
+    },
+    {
+      onError: (err) => {
+        const error = ensureError(err)
+        Logger.error(TAG, 'usePrepareDepositAndWithdrawTransactions - Error:', error)
+      },
+    }
+  )
+
+  return {
+    prepareTransactionsResult: prepareTransactions.result,
+    refreshPreparedTransactions: prepareTransactions.execute,
+    clearPreparedTransactions: prepareTransactions.reset,
+    prepareTransactionError: prepareTransactions.error,
+    isPreparingTransactions: prepareTransactions.loading,
+  }
+}
+
+export function usePrepareClaimExitAndWithdrawTransactions(
```

Comment:
Not a huge deal, but it's slightly strange to me that one of these functions (this one, `usePrepareClaimExitAndWithdrawTransactions`) actually performs the call on the TX preparer, and the other (`usePrepareDepositAndWithdrawTransactions`) just returns a callback fn + execution state. I wonder if the naming of `usePrepareDepositAndWithdrawTransactions` could be updated to reflect this, e.g. `usePrepareDepositAndWithdrawTransactionsCallback`.

Reply:
Renamed in 3de16cdb20bb644b0541382d5e4b517f4fede2ed.